### DISCUSSION TITLE
Next.js: console.log inside getStaticProps() for a file that's inside pages directory doesn't work

### DISCUSSION THREAD

**Discussion by nandramihnea** [vercel/next.js#40540] [Help] [answered]
I have a Next.js project setup.
I have tried to connect to MongoDB following step-by-step a very nice tutorial [(https://www.mongodb.com/developer/languages/javascript/nextjs-with-mongodb/)] but couldn't manage it because the data that I've tried to return through `props` was always undefined. Here is the code:
```
import { connectToDatabase } from 'helpers/mongodb'
import React, { useEffect } from 'react'

export default function Intrebari({ data }) {
  useEffect(() => {
    console.log('data', data);
  }, [data])

  return (
    <section>
      <div className="wrapper grid content-between">
        <span className="text-lg text-primary-9 font-medium mb-24">
          Întrebări frecvente
        </span>

        {data.map((item) => (
          <div key={item._id}>
            <p>Question: {item.question}</p>
            <p>Answer: {item.answer}</p>
          </div>
        ))}
      </div>
    </section>
  );
}

export async function getStaticProps() {
  const { db } = await connectToDatabase();
  console.log('db', db);

  const data = await db
    .collection("intrebari")
    .find({})
    .toArray();

  return {
    props: {
      data: JSON.parse(JSON.stringify(data)),
    },
  };
};
```

I then tried to `console.log` something inside `getStaticProps` but nothing was showing in the terminal.

```
export async function getStaticProps() {
  console.log('db');
  return {
    props: {
      data: JSON.parse(JSON.stringify(data)),
    },
  };
};
```

What can be the problem?
I repeat, the file that I'm using is inside the `pages` folder

**Comment by interglobalmedia** [comment]
Interesting. I had the same issue. And I am on 12.3.1, and React 18.2.0.

**Comment by interglobalmedia** [comment]
I figured it out. You probably added the console.log() after you had run the build script. If you re-run the script again, and then run npm start,the console.log should appear. That is exactly what I did now.

**Reply by nandramihnea** [reply]
![image](https://user-images.githubusercontent.com/23297718/195357066-617914c9-6596-436f-af15-4b1c32bc1cf9.png)
doesn't seem to be the case. I've ending the current build and re-build the script -> npm run dev but still no console.log displayed

**Comment by icyJoseph** [comment, marked as answer]
Do you maybe have:

```js
/** @type {import('next').NextConfig} */
const nextConfig = {
  reactStrictMode: true,
  compiler: {
    removeConsole: true,
  },
};

module.exports = nextConfig;
```

something like the above?

## Edit

Lazy generator of pages, or on demand generation, made it so that OP didn't directly see the message.

**Reply by nandramihnea** [reply]
```
const nextConfig = {
  reactStrictMode: true,
  swcMinify: true,
}
```
doesn't seem like this is the case

**Reply by icyJoseph** [reply]
can you try with like `console.count('abc')` or other console methods?

**Reply by nandramihnea** [reply]
yup, it's the same thing, nothing displayed. I've tried to re-build with console.count but I had the same issue, no log displayed in the terminal. Also, I'm very new on next.js this being my first try with the platform

**Reply by nandramihnea** [reply]
might be doing something wrong though because seems weird to me as well

**Reply by icyJoseph** [reply]
But there are logs on that terminal screenshot, like info and company, those are calls to `console.log`, so in principle logging is working. Is it possible to take a look at this repository? I hope it is not a VSCode-built-in-terminal + windows kind of problem, but rather something we are missing on the node/next.js side.

**Reply by icyJoseph** [reply]
You are navigating to the page right? Sorry if it sounds rude, but Next.js does lazy generation, meaning you have to navigate to the page to trigger any work at all.

**Reply by icyJoseph** [reply]
Sorry for spam, could you try with `console.log('\x1b[33m%s\x1b[0m', 'hello world'); ` that'll make the text yellow-ish, just trying to discard some kind of problem with powershell

**Reply by nandramihnea** [reply]
damn man. You need to know when to ask good questions. This was a brilliant one and made me feel good because my console.log problem is solved now and I learned something new. This was why I wasn't able to see the console.log, I wasn't navigating to the page itself.
Thank you!

**Reply by icyJoseph** [reply]
Nice! Next the Mongo DB issue 😅 Good luck!

**Reply by nandramihnea** [reply]
I'll try my best this evening :). Thanks for being helpful.
